base/uk.ac.stfc.isis.ibex.ui.devicescreens.tests/src/uk/ac/stfc/isis/ibex/ui/devicescreens/tests/DeviceScreensDescriptionViewModelTest.java

```diff
@@ -134,55 +135,97 @@ public void constructor_produces_deep_copy_of_description() {
     @Test
     public void changing_the_selected_screen_updates_name_key_etc() {
         // Act
-        viewModel.setSelectedScreen(1);
+        List<DeviceDescriptionWrapper> first = viewModel.getScreens().subList(1, 2);
+        viewModel.setSelectedScreens(first);
         
         // Assert
-        assertEquals(deviceName + "2", viewModel.getCurrentName());
+        assertEquals(deviceName + "2", viewModel.getName());
         assertEquals(opiName1, viewModel.getCurrentKey());
-        assertEquals(opiDescription1, viewModel.getCurrentDescription());
-        assertEquals(deviceName + "2", viewModel.getSelectedScreen().getName());
+        assertEquals(opiDescription1, viewModel.getDescription());
     }
 
     @Test
     public void clearing_the_selected_screen_unsets_name_key_etc() {
         // Act
-        viewModel.setSelectedScreen(-1);
+        viewModel.setSelectedScreens(null);
 
         // Assert
-        assertEquals("", viewModel.getCurrentName());
+        assertEquals("", viewModel.getName());
         assertEquals("", viewModel.getCurrentKey());
-        assertEquals("", viewModel.getCurrentDescription());
-        assertEquals(null, viewModel.getSelectedScreen());
+        assertEquals("", viewModel.getDescription());
+    }
+
+    @Test
+    public void setting_multiple_screens_unsets_name_key_etc() {
```

Comment:
I think the team standard is now to use GIVEN__WHEN__THEN naming. From the point of view of behaviour driven development, this test suggests that the name, key and description MUST be set to blank when multiple screens are selected. Is this true or has blank string been selected as a sensible option? Although you might leave the test content the same, make sure that the name reflects the desired behaviour, not the specific implementation.

This comment repeats as necessary for all other new tests.